cmd/resolve-version/main.go

```diff
@@ -307,19 +307,23 @@ func parseObject(key string) (release, error) {
 // Wrapper around the S3 API for listing objects
 // This maps directly to the API and parses the XML response but will not handle
 // paging and offsets automaticaly
-func fetchS3Result(bucketName string, options map[string]string) (result, error) {
+func fetchS3Result(bucketName string, region string, options map[string]string) (result, error) {
 	var result result
 	v := url.Values{}
 	v.Set("list-type", "2")
 	for key, val := range options {
 		v.Set(key, val)
 	}
-	url := fmt.Sprintf("https://%s.s3.amazonaws.com?%s", bucketName, v.Encode())
+	url := fmt.Sprintf("https://%s.s3.%s.amazonaws.com?%s", bucketName, region, v.Encode())
 	resp, err := http.Get(url)
 	if err != nil {
 		return result, err
 	}
 
+	if resp.StatusCode > 400 {
```

Comment:
Unless there's some redirect-handling I'm missing here, then 300's would result in invariant behavior too:

```suggestion
	if resp.StatusCode >= 300 {
```

Reply:
That seems reasonable to me

Reply:
Good catch on the `>=` too